### DISCUSSION TITLE
help

### DISCUSSION THREAD

**Discussion by MareSmaX** [arkypita/LaserGRBL#1197] [Q&A]
Hi everyone, I'm Giuseppe, I'm writing to everyone for a solution that for now I can't solve, I have a CNC 1310, and lately I decided to try to engrave with laser module 5500w, I made an update of the controller to the GRBL 1.1 version , I have inserted the laser in the appropriate jumber I connect it to the GRBLaser software, I enter all the parameters with the gcode that interests me, but when I click on start, the machine follows all the movements perfectly, but the laser does not emit any signal from the controller, someone can you help me on this problem? I have already enabled co $$ the command $ 31 = 1 but nothing starts nothing, for any info I am waiting, thanks in advance.

Salve a tutti, sono Giuseppe, scrivo a tutti per una soluzione che per adesso non riesco a risolvere, ho una CNC 1310, e ultimamente ho deciso di provare ad incidere con laser modulo da 5500w, ho fatto un aggiornamento del controller alla versione GRBL 1.1, ho inserito nell apposito jumber il laser lo collego al software GRBLaser, inserisco tutti i parametri con il gcode che mi interessa, ma quando clicco su avvia, la macchina segue tutti i movimenti perfettamente, ma il laser non emette nessun segnale dal controller, qualcuno puo' aiutarmi su questo problema? ho gia abilitato co $$ il comando $31=1 ma niente non si avvia nulla, per qualsiasi info sono in attesa, grazie in anticipo.

**Comment by belialboy** [comment]
Novice advice here, but I think $31 sets the Minimum Spindle Speed. $32 enables Laser-mode.  M3 and M5 commands turn the laser on and off, and the S value (0-255) sets the intensity of the laser from 0-100%.

**Comment by MareSmaX** [comment]
Thank you so much for the answer, I solved it by inserting the Y-cable, but today I have a problem that I still can't solve, after having prepared a written and created the Gcode, I send everything to start the processing, but in the end the writing comes to me mirror, but I still don't know how to fix this.

Grazie infinite per la risposta, ho risolto inserendo il cavo a Y, ma oggi ho un problema che ancora non riesco a risolvere, dopo aver preparato una scritta e creato il Gcode, invio tutto per avviare la lavorazione, ma alla fine la scritta mi viene a specchio, ma non so ancora come risolvere questo problema.

**Comment by StuartB4** [comment]
You may need to change $3 to invert the axis. 

![image](https://user-images.githubusercontent.com/16980026/104062041-998c2980-51f1-11eb-9050-d13ed67f7a47.png)


Questa impostazione inverte il segnale di direzione per ogni asse. Per impostazione predefinita, Grbl presume che gli assi si muovano in una direzione positiva quando il segnale del pin di direzione è basso e in una direzione negativa quando il pin è alto. Spesso gli assi non si muovono in questo modo con alcune macchine. Questa impostazione invertirà il segnale del perno di direzione per quegli assi che si muovono nella direzione opposta.

Questa impostazione della maschera di inversione funziona esattamente come la maschera di inversione della porta di passaggio e memorizza gli assi da invertire come flag di bit. Per configurare questa impostazione, è sufficiente inviare il valore per gli assi che si desidera invertire. Usa la tabella sopra. Ad esempio, se si desidera invertire solo la direzione dell'asse Y, inviare $ 3 = 2

**Comment by MareSmaX** [comment]
Thanks for the answer, yes I had already inverted the axes, but the problem is when writing in mirror mode, it does not write as created the Gcode, but everything in reverse
